lib/spack/spack/relocate.py

```diff
@@ -403,6 +403,10 @@ def _set_elf_rpaths(target, rpaths):
     # Join the paths using ':' as a separator
     rpaths_str = ':'.join(rpaths)
 
+    # patchelf segfaults when you pass it an empty rpath string.
+    if len(rpaths_str) == 0:
+        return ''
```

Comment:
I think this can only happen if `rpaths` is empty. Can we as a nitpick:
```suggestion
    if not rpaths:
        return ''
```
before even constructing the string (i.e. at line 403)?

Reply:
That doesn't work when passing `['']`

Reply:
Ugh, is that something that happens in Spack? In that case there's an error in the caller, which is supposed to pass paths and not an empty string (I thought the error here was not dealing with the empty list).

Reply:
https://github.com/spack/spack/pull/27675#issuecomment-981586786